client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -1161,6 +1161,26 @@ public void resetOffset(String topic, String group, Map<MessageQueue, Long> offs
         }
     }
 
+    public List<MessageQueue> getAllocationResult(final String topic, final String group, final String strategyName,
+        final List<MessageQueue> mqAll, final List<String> cidAll) {
+        String brokerAddr = this.findBrokerAddrByTopic(topic);
+        if (null == brokerAddr) {
+            this.updateTopicRouteInfoFromNameServer(topic);
+            brokerAddr = this.findBrokerAddrByTopic(topic);
+        }
+
+        if (null != brokerAddr) {
+            try {
+                return this.mQClientAPIImpl.getAllocationResultByStrategy(brokerAddr, group, clientId, strategyName,
+                    mqAll, cidAll, 3000);
+            } catch (Exception e) {
+                log.warn("getAllocationResultByStrategy exception, " + brokerAddr, " " + group, e);
```

Comment:
It would be better to use placeholders, such as `log.warn("getAllocationResultByStrategy exception, {} {}", brokerAddr, group, e);`

Reply:
Updated.